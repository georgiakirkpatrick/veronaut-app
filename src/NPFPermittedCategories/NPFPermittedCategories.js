import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'
import formData from '../FORM_DATA'

const noneOption = [
    {
        text: 'None of these',
        id: 101,
        checked: false
    }
]

const NPFPermittedCategories = props => {
    const {
        currentPage,
        none,
        noneChange,
        pageTurns,
        pCatChange,
        pCategories,
        setPage,
    } = props

    const nextButton = () => {
        const permCats = []

        Object.keys(pCategories).forEach(key => {
            if (pCategories[key]) {
                permCats.push(key)
            }
        })

        if (permCats.length === 0 && !none[101]) {
            alert('Please select an option.')
        } else {
            setPage(currentPage + pageTurns)
        }
    }

    return (
        <div className='NPFPermittedCategories'>
            <FormPage title='Categories'>
                <FormPromptWithSub
                    prompt='Does the product fall into one of the following categories?'
                    promptSubtitle="Check any that apply or 'none of these' and then 'next'."
                />

                <FormCheckboxSection 
                    options={formData.permittedCategories}
                    selectedOptions={pCategories}
                    handleChange={event => pCatChange(event)}
                />

                <FormCheckboxSection 
                    options={noneOption}
                    selectedOptions={none}
                    handleChange={event => noneChange(event)}
                />
            </FormPage>

            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => setPage(currentPage - 1)} 
                nextButton={() => nextButton()} 
            />
        </div>
    )
}

NPFPermittedCategories.defaultProps = {
    currentPage: 1,
    none: {},
    noneChange: () => {},
    pageTurns: () => {},
    pCatChange: () => {},
    pCategories: {},
    setPage: () => {}
}

export default NPFPermittedCategories