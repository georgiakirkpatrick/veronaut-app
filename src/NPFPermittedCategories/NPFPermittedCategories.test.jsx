import React from 'react'
import { render, screen } from '@testing-library/react'
import NPFPermittedCategories from './NPFPermittedCategories'
import { describe, it, expect } from 'vitest'

describe ('NPFPermittedCategories', () => {
    it ('renders without crashing', () => {
        render(<NPFPermittedCategories />)
        screen.debug()
    })

    it('renders the UI as expected', () => {
        const tree = render(<NPFPermittedCategories />)
        expect(tree).toMatchSnapshot()    
    })
})