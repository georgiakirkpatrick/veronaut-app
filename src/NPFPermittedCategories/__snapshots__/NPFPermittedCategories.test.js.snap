// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFPermittedCategories renders the UI as expected 1`] = `
<div>
  <section
    className="FormPage"
  >
    <h1>
      Categories
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Does the product fall into one of the following categories?
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          Check any that apply or 'none of these' and then 'next'.
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <div
          className="empty"
        />
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={1}
              name={1}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={1}
            >
              Outerwear
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={2}
              name={2}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={2}
            >
              Shoes
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={3}
              name={3}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={3}
            >
              Socks and tights
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={4}
              name={4}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={4}
            >
              Swimwear
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={5}
              name={5}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={5}
            >
              Underwear
            </label>
          </div>
        </div>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <div
          className="empty"
        />
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={101}
              name={101}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={101}
            >
              None of these
            </label>
          </div>
        </div>
      </div>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
